proposals/low-level-struct-improvements.md

```diff
@@ -263,6 +262,16 @@ ref int Sneaky(out int i)
 }
 ```
 
+Languages that support `ref struct` must ensure the original value passed into an `out` parameter for a `ref struct` is not directly returned or indirectly through another `ref` just as they would for a `ref scoped` parameter. Doing so would allow for type safety holes to exist. C# achieves this via it's definite assignment rules. That both achieves our ref safety as well as allowing for existing code which assigns and then returns `out` parameters values.
+
+```c#
+ref Span<int> StrangeButLegal(out Span<int> span)
+{
+    span = default;
+    return span;
```

Comment:
>return span; [](http://example.com/codeflow?start=4&length=12)

The method signature returns a `ref` rather than a value. Also, the `out Span<int> span` cannot be returned by `ref` nor by value since it is `out scoped`.